src/resources/platform_handlers/pytorch_libtorch/model.py

```diff
@@ -0,0 +1,238 @@
+        """
+        self._model_name = args["model_name"]
+        self._logger = pb_utils.Logger
+        self._logger.log_info(
+            "Initializing model instance for '" + self._model_name + "'"
+        )
+
+        self._model_config = json.loads(args["model_config"])
+        self._inputs = _parse_io_config(self._model_config["input"])
+        self._outputs = _parse_io_config(self._model_config["output"])
+
+        kind = args["model_instance_kind"]
+        device_id = args["model_instance_device_id"]
+        self._device = torch.device(_get_device_name(kind, device_id))
+
+        model_path = _get_model_path(self._model_config)
+        if not _is_py_class_model(model_path):
+            self._logger.log_info("Loading '" + self._model_name + "' as TorchScript")
+            self._model = torch.jit.load(model_path)
+            self._model.to(self._device)
+            self._model.eval()
+            return
+
+        self._model_module = _import_module_from_path(self._model_name, model_path)
+        self._model_class = _get_model_class_from_module(self._model_module)
+        self._raw_model = self._model_class()
+        data_path = _get_model_data_path(model_path)
+        if data_path != "":
+            self._raw_model.load_state_dict(
+                torch.load(data_path, map_location=self._device)
+            )
+        else:
+            self._logger.log_info(
+                "Model parameter file not found for '" + self._model_name + "'"
+            )
+        self._raw_model.to(self._device)
+        self._raw_model.eval()
+        if _enable_torch_compile(self._model_config):
+            self._model = torch.compile(self._raw_model)  # PyTorch 2.0+ only
+        else:
+            self._logger.log_info(
+                "'torch.compile' is disabled for '" + self._model_name + "'"
+            )
+            self._model = self._raw_model
+
+    def execute(self, requests):
+        """`execute` MUST be implemented in every Python model. `execute`
+        function receives a list of pb_utils.InferenceRequest as the only
+        argument. This function is called when an inference request is made
+        for this model. Depending on the batching configuration (e.g. Dynamic
+        Batching) used, `requests` may contain multiple requests. Every
+        Python model, must create one pb_utils.InferenceResponse for every
+        pb_utils.InferenceRequest in `requests`. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        list
+          A list of pb_utils.InferenceResponse. The length of this list must
+          be the same as `requests`
+        """
+        responses = []
+
+        # FIXME: Instead of iterating through each request, run the inference as a single batch.
```

Comment:
Could you please clarify for me, is there a JIRA ticket for this (i.e. is there a feature request).

Reply:
I cannot find the ticket on JIRA. @tanmayv25 @Tabrizian Do you know if we are going to provide `requests` as batched tensor(s) instead of having to iterate over each `request` on a batch, in the future?

Reply:
@kthui  The python backend side ticket is here: DLIS-2873

Till we add this enhnacement on the python backend, I would suggest we can modify the implementation here itself. We can merge the request buffers and obtain batched result with a single model execution. This can be followed by splitting the result into their multiple response objects.

Reply:
Yes. The ability to batch requests into a single tensor is added.